### PR TITLE
Add AI Runtime exist model check

### PR DESCRIPTION
**Pull request by brosoul**
## Pull Request Description
Add AI Runtime exist model check
- Before downloading the file, check the meta file under `.cache` to verify if the file already exists
- add `DOWNLOADER_FORCE_DOWNLOAD` and `DOWNLOADER_CHECK_FILE_EXIST` to control whether the above verification logic is enabled or not

Example:
First time downloading files from S3
image

Download the file again from S3 to the same path
image

## Related Issues
Resolves: #149 

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: python/aibrix/aibrix/downloader/tos.py [hunk 1]
```diff
@@ -100,15 +103,34 @@ def download(
...
-        # TOS client does not support Path, convert it to str
-        local_file = str(local_path.joinpath(_file_name).absolute())
+        local_file = local_path.joinpath(_file_name).absolute()
+
+        # check if file exist
+        etag = meta_data.etag
+        file_size = meta_data.content_length
+        meta_data_file = meta_file(local_path=local_path, file_name=_file_name)
+
+        if not envs.DOWNLOADER_FORCE_DOWNLOAD and envs.DOWNLOADER_CHECK_FILE_EXIST:
```
Comment (Jeffwan) [inline python/aibrix/aibrix/downloader/tos.py]: seems the logic here are same as s3? is it better to extract a base method to reuse?

File: python/aibrix/aibrix/config.py [hunk 2]
```diff
@@ -14,3 +14,5 @@
 
 
 DEFAULT_METRIC_COLLECTOR_TIMEOUT = 1
+
+DOWNLOAD_CACHE_DIR = ".cache"
```
Comment (Jeffwan) [inline python/aibrix/aibrix/config.py:18]: this is a relative path? for every model weights, we have a cache dir, right?

File: python/aibrix/aibrix/downloader/utils.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from pathlib import Path
+from typing import Union
+
+from aibrix.config import DOWNLOAD_CACHE_DIR
+
+
+def meta_file(local_path: Union[Path, str], file_name: str) -> Path:
```
Comment (Jeffwan) [inline python/aibrix/aibrix/downloader/utils.py:26]: let's add some unit test for all utils in this file

File: python/aibrix/aibrix/config.py (hunk 2 above)
Comment (brosoul) [reply python/aibrix/aibrix/config.py:18]: Yes

File: python/aibrix/aibrix/downloader/utils.py (hunk 3 above)
Comment (brosoul) [reply python/aibrix/aibrix/downloader/utils.py:26]: done

**Pull request merged**